httpclient/response.go

```diff
@@ -55,6 +55,18 @@ func WithResponseHeader(key string, value *string) DoOption {
 
 func WithResponseUnmarshal(response any) DoOption {
 	return DoOption{
+		in: func(r *http.Request) error {
+			if r.Header.Get("Accept") != "" {
+				return nil
+			}
+			switch response.(type) {
+			case *bytes.Buffer, *io.ReadCloser, *[]byte:
+				r.Header.Set("Accept", "application/octet-stream")
```

Comment:
Ok, let's just do nothing in this case then. Again, I would like to minimize coupling between Go types and request behavior where it isn't implied.

Reply:
fine...